### PR TITLE
Little refactoring

### PR DESCRIPTION
**Pull request by ostark**
Please review. And let's discuss.

### CODE REVIEW

File: src/ElementStatusChange.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+     * @param \yii\base\Application $app
+     */
+    public function bootstrap($app)
+    {
+        // Make sure it's Craft
+        if (!($app instanceof CraftWebApp || $app instanceof CraftConsoleApp)) {
+            return;
+        }
+
+        Event::on(Elements::class, Elements::EVENT_BEFORE_SAVE_ELEMENT, [$this, 'rememberPreviousStatus']);
```
Comment (ostark) [inline src/ElementStatusChange.php:52]: > Why using a handler method over a closure?

For readability - the method name clearly indicates what the handler does.

File: src/behaviors/ElementStatusBehavior.php [hunk 2]
```diff
@@ -23,18 +19,14 @@ class ElementStatusBehavior extends Behavior
      */
     public $statusBeforeSave = '';
 
-    /**
-     * @var bool
-     */
-    public $statusChanged = false;
```
Comment (ostark) [inline src/behaviors/ElementStatusBehavior.php:29]: > Why removing this property?

The event is only fired IF the status has changed.

File: src/behaviors/ElementStatusBehavior.php [hunk 3]
```diff
@@ -23,18 +19,14 @@ class ElementStatusBehavior extends Behavior
...
 
     // Public Methods
     // =========================================================================
 
     /**
      * Saves the status of an element before it is saved
      */
-    public function onBeforeSaveStatus()
+    public function rememberPreviousStatus()
```
Comment (ostark) [inline src/behaviors/ElementStatusBehavior.php:29]: > Why renaming the method names?
> `onBeforeSaveStatus` to `rememberPreviousStatus`
> `onAfterSaveStatus` to `fireEventOnChange`

From outside it's clear what's happening inside. Compare:

`$element->onBeforeSaveStatus()` vs `$element->rememberPreviousStatus()`
`$element->onAfterSaveStatus()` vs `$element->fireEventOnChange()`

File: src/behaviors/ElementStatusBehavior.php [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
 
 class ElementStatusBehavior extends Behavior
 {
-    // Constants
-    // =========================================================================
-
-    /**
-     * @event Event
-     */
-    const EVENT_STATUS_CHANGED = 'statusChanged';
```
Comment (ostark) [inline src/behaviors/ElementStatusBehavior.php:16]: > Why moving the event name from the Behavior to the main class?

The event name is the "interface" for the consumer of the library. The Behavior is an implementation detail he/she should not worry about. And, decoupling event from the Behavior allows us to fire the event independently. E.g. in 
`commands/ScheduledElements::fireEvent()`
https://github.com/putyourlightson/craft-element-status-events/pull/1/files#diff-0068c7f8d063b0e3caf057549ad8a6d9R113

File: composer.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
 {
     "name": "putyourlightson/craft-element-status-events",
-    "description": "Element status events module for Craft CMS.",
-    "version": "1.2.0",
-    "type": "craft-module",
+    "description": "Element status events extension for Craft CMS.",
+    "version": "2.0.0",
```
Comment (ostark) [inline composer.json:4]: > Why the version bump?

The chance is low somebody uses this library so far, however changing the "public interface" introduces breaking changes for the consumer. According to SemVer this is only possible in a MAJOR release.

File: composer.json [hunk 6]
```diff
@@ -13,12 +13,24 @@
...
+            "name": "Ben Croker",
+            "homepage": "https://putyourlightson.com",
+            "role": "Developer"
+        },
+        {
+            "name": "Oliver Stark",
+            "homepage": "https://www.fortrabbit.com",
+            "role": "Developer"
+        }
+    ],
```
Comment (ostark) [inline composer.json:27]: > Fair?

File: composer.json [hunk 7]
```diff
@@ -27,12 +39,6 @@
...
     },
     "extra": {
-        "name": "Element Status Events",
-        "handle": "element-status-events",
-        "developer": "PutYourLightsOn",
-        "developerUrl": "https://putyourlightson.com/",
-        "schemaVersion": "1.0.0",
-        "changelogUrl": "https://raw.githubusercontent.com/putyourlightson/craft-element-status-events/v1/CHANGELOG.md",
-        "class": "putyourlightson\\elementstatusevents\\ElementStatusEvents"
+        "bootstrap": "putyourlightson\\elementstatusevents\\ElementStatusChange"
```
Comment (ostark) [inline composer.json:42]: > Most of these extra settings are Craft Plugin specific, but it's not a plugin

File: src/ElementStatusChange.php [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * Class ElementStatusChange
+ *
+ * @package putyourlightson\elementstatusevents
+ */
+class ElementStatusChange extends Component implements BootstrapInterface
+{
+
+    const EVENT_STATUS_CHANGED = 'statusChanged';
+
```
Comment (ostark) [inline src/ElementStatusChange.php:28]: > Why renaming the main class?

The class + event constant is the public API for the consumer.
Detecting the "status **change**" is the only purpose of the package. Being more specfic (`ElementStatusChange` vs `ElementStatusEvents`) felt bit better to me. 
However, I'm not super opinionated about that.

**Review by ostark — COMMENTED** [review]
More explainations

**Comment by ostark** [comment]
ping

**Comment by putyourlightson** [comment]
Pong. Thanks for the follow-up, still on my list to review, just need to find the time to look at it all in detail :)

File: composer.json (hunk 6 above)
Comment (putyourlightson) [reply composer.json:27]: Absolutely!

**Comment by putyourlightson** [comment]
All looks good Oliver, I'll resolve the conflicts so I can merge and then perhaps make a few minor tweaks. I really like how this has evolved into something very clean and readable.

**Comment by putyourlightson** [comment]
Merged into the v2 branch.

**Comment by putyourlightson** [comment]
I changed the name of the main class back to `ElementStatusEvents` so it matches the name of the repo.

**Pull request closed without merging**